projects/packages/search/search.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Search global definitions and operations.
+ *
+ * @package automattic/jetpack-search
+ */
+
+namespace Automattic\Jetpack\Search;
+
+define( 'JETPACK_SEARCH_PKG__VERSION', '0.6.0-alpha' );
+define( 'JETPACK_SEARCH_PKG__DIR', __DIR__ );
+define( 'JETPACK_SEARCH_PKG__SLUG', 'jetpack-search-pkg' );
+
+// If WordPress's plugin API is available already, use it. If not,
+// drop data into `$wp_filter` for `WP_Hook::build_preinitialized_hooks()`.
+if ( function_exists( 'add_filter' ) ) {
+	$add_filter = 'add_filter';
+	// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+	$add_action = 'add_action';
+} else {
+	$add_filter = function ( $name, $cb, $priority = 10, $accepted_args = 1 ) {
+		global $wp_filter;
+		// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+		$wp_filter[ $name ][ $priority ][] = array(
+			'accepted_args' => $accepted_args,
+			'function'      => $cb,
+		);
+	};
+	// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+	$add_action = $add_filter;
+}
+
+/**
+ * Adds the package slug and version to the package version tracker's data.
+ *
+ * @param array $package_versions The package version array.
+ *
+ * @return array The packge version array.
+ */
+function send_version_to_tracker( $package_versions ) {
+	$package_versions[ JETPACK_SEARCH_PKG__SLUG ] = JETPACK_SEARCH_PKG__VERSION;
+	return $package_versions;
+}
+
+// Set up package version hook.
+$add_filter( 'jetpack_package_versions', __NAMESPACE__ . '\send_version_to_tracker' );
```

Comment:
In other packages such as Backup or Connection, we set up a new class just for this, in `projects/packages/connection/src/class-package-version.php` and `projects/packages/backup/src/class-package-version.php`. Would it be worth following a similar pattern for the Search package?

cc @fgiannar and @kbrown9, who worked on package version, and package version tracking for Jetpack packages, and who may have more insight.

Reply:
Yep didn't want to add a new class which only has one function, and ended up with this approach. Can move it to a class too. Any suggestions appreciated 😄

Reply:
If I remember correctly, I created those small package version classes just because I didn't think it made sense to add the package version constant and tracker hook callback to the existing package classes. I don't see any problems with this approach.

Will you be using the search package version on WPCOM? If not, it's not necessary to add it to the package version tracker. If you will be using the search package version on WPCOM, we might want to make the package slug similar to what we're using for the other packages. We use `connection`, `sync`, and `backup`, so maybe this could just be `search`.

Reply:
Hi @kbrown9 Thanks for providing the helpful information. Just FYI The package to be used on WPCOM as well, so probably we'd need the version to be reported. Also I'll change to slug to `search`, thanks for the suggestions.

Reply:
addressed in 96d25aeadc